src/Renci.SshNet/ISftpClient.cs

```diff
@@ -211,7 +212,7 @@ public interface ISftpClient : IBaseClient
         /// <remarks>
         /// Method calls made by this method to <paramref name="output" />, may under certain conditions result in exceptions thrown by the stream.
         /// </remarks>
-        IAsyncResult BeginDownloadFile(string path, Stream output, AsyncCallback asyncCallback, object state, Action<ulong> downloadCallback = null);
+        IAsyncResult BeginDownloadFile(string path, Stream output, AsyncCallback asyncCallback, object? state, Action<ulong>? downloadCallback = null);
```

Comment:
Every `AsyncCallback` in this file and in SftpClient.cs can be nullable

```suggestion
        IAsyncResult BeginDownloadFile(string path, Stream output, AsyncCallback? asyncCallback, object? state, Action<ulong>? downloadCallback = null);
```